presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoPreparedStatement.java

```diff
@@ -131,11 +133,59 @@ public boolean execute()
         return super.execute(getExecuteSql());
     }
 
+    @Override
+    public int[] executeBatch() throws SQLException
+    {
+        if (!isInsertValuesWithParameter()) {
+            throw new SQLException("Execute batch is only supported for insert values with parameter statement");
+        }
+        String batchExecuteSql = getBatchExecuteSql();
+        // if there is no parameter set, return an empty array directly
+        if (batchExecuteSql == null) {
+            return new int[] {};
+        }
+        super.executeBatch(batchExecuteSql);
+        ResultSet rs = getResultSet();
+
+        // executeBatch() would get a result of Array[Integer] type from server, and transform it to a int[]
+        int[] res = new int[0];
+        while (rs.next()) {
```

Comment:
This `rs` should just return one result, right?

Reply:
Yes, that's right.